Source/Events.Store/IFetchCommittedEvents.cs

```diff
@@ -40,6 +42,26 @@ public interface IFetchCommittedEvents
     /// <returns>A <see cref="Task" /> that, when resolved, returns the <see cref="CommittedAggregateEvents"/> containing all <see cref="CommittedAggregateEvent"/> applied to the Event Source by the Aggregate root, in the order of which they appear in the Event Log.</returns>
     Task<CommittedAggregateEvents> FetchForAggregate(EventSourceId eventSource, ArtifactId aggregateRoot, CancellationToken cancellationToken);
 
+    /// <summary>
+    /// Fetches all <see cref="CommittedAggregateEvent"/>s applied to an Event Source by an Aggregate Root.
+    /// </summary>
+    /// <param name="eventSource">The <see cref="EventSourceId"/> identifying the Event Source.</param>
+    /// <param name="aggregateRoot">The <see cref="ArtifactId"/> identifying the Aggregate Root.</param>
+    /// <param name="eventTypes">The <see cref="IEnumerable{T}"/> of <see cref="Artifact"/> event types of the aggregate events that should be fetched.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken" />.</param>
+    /// <returns>A <see cref="Task" /> that, when resolved, returns the a stream of <see cref="CommittedAggregateEvents"/> containing all <see cref="CommittedAggregateEvent"/> applied to the Event Source by the Aggregate root, in the order of which they appear in the Event Log that have one of the requested event types.</returns>
+    Task<Try<(AggregateRootVersion AggregateRootVersion, IAsyncEnumerable<CommittedAggregateEvent> EventStream)>> FetchStreamForAggregate(EventSourceId eventSource, ArtifactId aggregateRoot, IEnumerable<Artifact> eventTypes, CancellationToken cancellationToken);
```

Comment:
I don't know if this would make the API simpler or worse - but just a thought. These two streaming methods could respond with an `IAsyncEnumerable<CommittedAggregateEvents>` instead of a version + stream of events.

It could either do some batching internally - or just send out a stream with a single event in each. It feels like this could simplify the conversions that now happen outside of this interface a little bit. What do you think?

Reply:
I'm not sure. The reason for the `AggregateRootVersion` at least is for knowing the current version of the aggregate root. But we could turn this a bit around as you said and have this return the batches